packages/smithy-core/src/smithy_core/traits.py

```diff
@@ -307,3 +313,9 @@ def host_prefix(self) -> str:
 class HostLabelTrait(Trait, id=ShapeID("smithy.api#hostLabel")):
     def __post_init__(self):
         assert self.document_value is None
+
+
+@dataclass(init=False, frozen=True)
+class RestJson1Trait(Trait, id=ShapeID("aws.protocols#restJson1")):
```

Comment:
This has a bunch of properties you'll need to add in here. Also, I vote that this definitely goes in the AWS package.